### PR TITLE
[Generate] Remove outdated code

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

This PR refactors `greedy_search()` and `sample()` by removing old code and improving some comments.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/generation_utils.py [hunk 1]
```diff
@@ -483,31 +483,6 @@ def _expand_inputs_for_generation(
...
 
-    @staticmethod
-    def _init_sequence_length_for_generation(
-        input_ids: torch.LongTensor, max_length: int
-    ) -> Tuple[torch.Tensor, torch.Tensor, int]:
-        unfinished_sequences = input_ids.new(input_ids.shape[0]).fill_(1)
-        sequence_lengths = input_ids.new(input_ids.shape[0]).fill_(max_length)
-
-        cur_len = input_ids.shape[-1]
-        return sequence_lengths, unfinished_sequences, cur_len
```
Comment (patrickvonplaten) [inline src/transformers/generation_utils.py:494]: `sequence_lengths` is never actually used anywhere so we can delete it completely

File: src/transformers/generation_utils.py [hunk 2]
```diff
@@ -483,31 +483,6 @@ def _expand_inputs_for_generation(
             model_kwargs["encoder_outputs"] = encoder_outputs
         return input_ids, model_kwargs
 
-    @staticmethod
-    def _init_sequence_length_for_generation(
```
Comment (patrickvonplaten) [inline src/transformers/generation_utils.py:487]: This function is only used twice and essentially boils down to a single line -> therefore it doesn't not merit to be factored out into a function.

File: src/transformers/generation_utils.py [hunk 3]
```diff
@@ -483,31 +483,6 @@ def _expand_inputs_for_generation(
...
-        input_ids: torch.LongTensor, max_length: int
-    ) -> Tuple[torch.Tensor, torch.Tensor, int]:
-        unfinished_sequences = input_ids.new(input_ids.shape[0]).fill_(1)
-        sequence_lengths = input_ids.new(input_ids.shape[0]).fill_(max_length)
-
-        cur_len = input_ids.shape[-1]
-        return sequence_lengths, unfinished_sequences, cur_len
-
-    @staticmethod
-    def _update_seq_length_for_generation(
```
Comment (patrickvonplaten) [inline src/transformers/generation_utils.py:497]: After deleting unused variable `sequence_lengths`, this function boils down to a single line which then does not need to be factored out into a function

File: src/transformers/generation_utils.py [hunk 4]
```diff
@@ -1271,10 +1246,9 @@ def greedy_search(
                 model_kwargs["encoder_outputs"].get("hidden_states") if output_hidden_states else None
             )
 
-        # init sequence length tensors
```
Comment (patrickvonplaten) [inline src/transformers/generation_utils.py:1274]: comment is not understandable

File: src/transformers/generation_utils.py [hunk 5]
```diff
@@ -1330,29 +1304,23 @@ def greedy_search(
...
-                sequence_lengths, unfinished_sequences = self._update_seq_length_for_generation(
-                    sequence_lengths, unfinished_sequences, cur_len, next_tokens == eos_token_id
-                )
-
-            # update model kwargs
             model_kwargs = self._update_model_kwargs_for_generation(
                 outputs, model_kwargs, is_encoder_decoder=self.config.is_encoder_decoder
             )
-
-            # increase cur_len
```
Comment (patrickvonplaten) [inline src/transformers/generation_utils.py:1352]: unnecessary comment IMO

**Review by Narsil — APPROVED** [review]
LGTM

**Review by patil-suraj — APPROVED** [review]
Looks good!

**Comment by patrickvonplaten** [comment]
cc @LysandreJik, @sgugger

**Pull request merged**